cluster-autoscaler/proposals/pricing.md

```diff
@@ -0,0 +1,177 @@
+ 
+* [I1] Doesn’t go 2 times up with a small change in absolute terms (100->50 mill cpu), is more or less constant for small pods. 
+* [I2] Still penalizes node types that have some completely unneeded resources (GPU). 
+ 
+C/T can be stabilized by adding some value X to C and T. Lets call it a big cluster damper. 
+So the formula is like (C+X)/(T+X). 
+ 
+Lets see what happens if X is the cost of running a 0.5 cpu pod and we have 1 pending pod of size 0.1 cpu. The preferred node
+is N1-standard-8.
+ 
+| Machine type | Calculation | Rank |
+|--------------|-------|------| 
+| N1-standard-2      |            (0.095 + 0.016) / (0.003 + 0.016) | 5.84  |
+| N1-standard-8      |          (0.380 + 0.016) / (0.003 + 0.016)   | 20.84  |
+| N1-standard-2+GPU  |       (0.795 + 0.016) / (0.003 + 0.016)      | 42  |
+ 
+And what if 1.5 cpu.
+ 
+| Machine type | Calculation | Rank |
+|--------------|-------|------| 
+| N1-standard-2     |          (0.095 + 0.016) / (0.003*15 + 0.016)| 1.81 |
+| N1-standard-8     |         (0.380 + 0.016) / (0.003*15 + 0.016) |  6.49 |
+| N1-standard-2+GPU |        (0.795 + 0.016) / (0.003*15 + 0.016)  | 13.0  |
+ 
+Slightly better, but still hard to combine linearly with NodeUnfitness being equal 1 or 4.
+ 
+Let’s try something different: 
+```
+NodeUnfitness*(C + X)/(T+X)
+```
+0.1 cpu request:
+
+| Machine type | Calculation | Rank |
+|--------------|-------|------| 
+| N1-standard-2  |           4 * (0.095 + 0.016) / (0.003 + 0.016) | 23.36 |
+| N1-standard-8  |         1 * (0.380 + 0.016) / (0.003 + 0.016) | 20.84 |
+| N1-standard-2+GPU |  4 * (0.795 + 0.016) / (0.003 + 0.016) | 168.0 |
+ 
+1.5 cpu request:
+
+| Machine type | Calculation | Rank |
+|--------------|-------|------| 
+| N1-standard-2 |           4 * (0.095 + 0.016) / (0.003*15 + 0.016) | 7.24 |
+| N1-standard-8 |          1 * (0.380 + 0.016) / (0.003*15 + 0.016) |  6.49 |
+| N1-standard-2+GPU |   4 * (0.795 + 0.016) / (0.003*15 + 0.016) | 52 |
+ 
+Looks better. So we are able to promote having bigger nodes if needed. However, what if we were to create 50 N1-standard-8 nodes to accommodate 50 x 1.5 cpu pods with strict PodAntiAffinity? Well, in that case we should probably go for n1-standard-2 nodes, however the above formula doesn’t promote that, because it considers the node unfit. So when requesting a larger number of nodes (in a single scale-up) we should probably suppress NodeUnfitness a bit. The suppress function should reduce the effect of NodeUnfitness when there is a good reason to do it. One of the good reasons is that the other option is significantly cheaper. In general the more nodes we are requesting the bigger the price difference can be. And if we are requesting just a signle node
+then this node should be well fitted to the cluster (than to the pod) so that other pods can also use it and the cluster administrator has less types of nodes to focus on. 
+ 
+We are looking for a function suppress(NodeUnfitness, NodeCount) that:
```

Comment:
Can you please explicitly define NodeCount as the number of nodes we're requesting (delta) and not the total number of nodes in the cluster? I agree it's logical/can be deduced, especially considering earlier assumption that in a larger cluster we'd generally prefer to have larger nodes, but isn't immediately obvious when reading this for the first time.